syntax = "proto3";

service Banking {
    rpc CreateAccount (CreateRequest) returns (CreateReply) {};
    rpc Add (AddRequest) returns (AddReply) {};
    rpc AddWOReturn (AddWORequest) returns (AddWOReply) {};
    rpc Sub (SubRequest) returns (SubReply) {};
}

message CreateRequest {
    string name = 1;
    int32 money = 2;
}

message CreateReply {
    string message = 1;
}

message AddRequest {
    string name = 1;
    int32 money = 2;
}

message AddReply {
    string message = 1;
}

message AddWORequest {
    string name = 1;
    int32 money = 2;
}

message AddWOReply {
}

message SubRequest {
    string name = 1;
    int32 money = 2;
}

message SubReply {
    string message = 1;
}